packages/app/retainer/retainer.go

```diff
@@ -177,11 +200,30 @@ func (r *Retainer) setupEvents() {
 	}, r.workerPool)
 
 	event.Hook(r.protocol.Events.Engine.EvictionState.EpochEvicted, r.storeAndEvictEpoch)
+
+	r.protocol.Engine().NotarizationManager.Events.EpochCommitted.AttachWithWorkerPool(event.NewClosure(func(e *notarization.EpochCommittedDetails) {
+		if cc := r.createOrGetCachedCommitment(e.Commitment); cc != nil {
+			var (
+				blockIDs = make(models.BlockIDs)
+				txIDs    = utxo.NewTransactionIDs()
+			)
+			_ = e.AcceptedBlocks.Stream(func(key models.BlockID) bool {
+				blockIDs.Add(key)
+				return true
+			})
+			_ = e.AcceptedTransactions.Stream(func(key utxo.TransactionID) bool {
+				txIDs.Add(key)
+				return true
+			})
+
+			cc.setCommitment(e.Commitment, blockIDs, txIDs, e.CreatedOutputs, e.SpentOutputs)
+		}
+	}), r.workerPool)
```

Comment:
I think this could be a separate WP with only 1 worker specifically processing the `EpochCommitted` event